proto/metrics/agent_payload.proto

```diff
@@ -15,6 +15,33 @@ message CommonMetadata {
 	string api_key = 6;
 }
 
+enum OriginProduct {
+	no_origin_product = 0;
+	reserved 1 to 16;
+}
+
+enum OriginCategory {
+	no_origin_category = 0;
+	reserved 1 to 16;
+}
+
+enum OriginService {
+	no_origin_service = 0;
+	reserved 1 to 216;
+}
+
+message Origin {
+	reserved 1 to 3;
+	OriginProduct origin_product = 4;
```

Comment:
I'm tempted to specify the 3 as `int32` fields instead of `OriginProduct`/`OriginCategory` since the contents are reserved anyway

Reply:
Agreed.

Reply:
I changed them to uint32 as Vector does https://github.com/vectordotdev/vector/blob/32245e0e760e70f5a1746c93fac25149aba13cfc/proto/dd_metric.proto#L16-L21

It should not make a difference (from [protobuf docs](https://protobuf.dev/programming-guides/proto3/#updating), also applies to `proto2`):
> int32, uint32, int64, uint64, and bool are all compatible – this means you can change a field from one of these types to another without breaking forwards- or backwards-compatibility. If a number is parsed from the wire which doesn’t fit in the corresponding type, you will get the same effect as if you had cast the number to that type in C++ (for example, if a 64-bit number is read as an int32, it will be truncated to 32 bits).

I would feel more comfortable with `int32` but also want to be compatible with Vector. Thoughts?

Reply:
I don't have a preference either way, though I guess `uint32` makes sense the fields can/should never be negative. Any reason why you'd prefer an `int32`?

Reply:
> I don't have a preference either way, though I guess uint32 makes sense the fields can/should never be negative. Any reason why you'd prefer an int32?

Just that protobuf enums are `int32`s under the hood (at least in Go), so it would be more consistent with the backend's proto, which uses an enum.